KProcessHacker/thread.c

```diff
@@ -283,13 +268,29 @@ NTSTATUS KpiOpenThreadProcess(
     if (!NT_SUCCESS(status))
         goto CleanupExit;
 
+    status = ObReferenceObjectByHandle(
+        ThreadHandle,
+        0,
+        *PsThreadType,
+        AccessMode,
+        &thread,
+        NULL
+        );
+
+    if (!NT_SUCCESS(status))
+    {
+        thread = NULL;
+        goto CleanupExit;
+    }
+    
+    // Note: Windows 7 and Windows 8 require KernelMode (dmex)
     status = ObOpenObjectByPointer(
         PsGetThreadProcess(thread),
         0,
         NULL,
         DesiredAccess,
         *PsProcessType,
-        AccessMode,
+        KernelMode,
```

Comment:
A note for observers. This is an intentional decision to bypass access checks allowing for inspection of the thread. This aligns this path with the existing code: https://github.com/processhacker/processhacker/blob/master/KProcessHacker/thread.c#L164

The call is restricted by the key validation and stricter yet for write access requiring domination and caller validation.